obm-autoset/applyobmsetting.py

```diff
@@ -0,0 +1,350 @@
+    if arg_username=="None":
+        for creds in OBM_CONFIG['bmc']:
+            if remote_shell('ipmitool -I lanplus -H ' + bmcip+' -U ' + creds['username']+' -P '+ creds['password'] + ' fru')['exitcode'] == 0:
+                usr = creds['username']
+                pwd = creds['password']
+                break
+    else:
+        usr=arg_username
+        pwd=arg_password
+    # Put the credential to OBM settings
+    if  usr!="":
+        payload = { "service": "ipmi-obm-service","config": {"host": bmcip, "user": usr,"password": pwd},"nodeId": nodeid}
+        api_data = rackhdapi("/api/2.0/obms", action='put', payload=payload)
+        if api_data['status']< 300:
+            return True
+    return False
+
+def node_select():
+    nodelist = []
+    if NODEID != 'None':
+        nodelist.append(NODEID)
+        return nodelist
+    else:
+        catalog = rackhdapi('/api/2.0/nodes')
+        if catalog['status'] != 200:
+            print '**** Unable to retrieve node list via API.\n'
+            sys.exit(255)
+        for nodeentry in catalog['json']:
+           if nodeentry['type'] == 'compute':
+               nodelist.append(nodeentry['id'])
+    if VERBOSITY >= 6:
+        print "Node List:"
+        print nodelist, '\n'
+    if len(nodelist) == 0:
+        print '**** Empty node list.\n'
+    return nodelist
+
+
+ARG_PARSER = argparse.ArgumentParser(description="Command Help")
+ARG_PARSER.add_argument("-user", default="None",help="User Name, default:None")
+ARG_PARSER.add_argument("-password", default="None",help="Password, default:None")
+ARG_PARSER.add_argument("-ora", default="localhost",help="Specify RackHD ora IP address, default:localhost")
+ARG_PARSER.add_argument("-v", default=0,type=int,help="Verbosity level of console output, default:0")
+ARG_PARSER.add_argument("-nodeid", default="None",help="You can specify one node to apply OBM setting, add Node ID here, default:None")
+ARG_PARSER.add_argument("-auth", default="off",choices=["off","on"],help="Specify if authentication is on , default:off")
+
+CMD_ARGS = vars(ARG_PARSER.parse_args())
+arg_username= CMD_ARGS["user"]
+arg_password=CMD_ARGS["password"]
+arg_ora=CMD_ARGS["ora"]
+VERBOSITY=CMD_ARGS["v"]
+NODEID=CMD_ARGS["nodeid"]
+
+if CMD_ARGS["auth"]=="off":
+    API_PORT =  str(OBM_CONFIG['ports']['http'])
+    API_PROTOCOL =  'http'
+else:
+    API_PORT = str(OBM_CONFIG['ports']['https'])
+    API_PROTOCOL = "https"
+AUTH_TOKEN = "None"
+REDFISH_TOKEN = "None"
+
+if arg_ora=="localhost":
+    print "Attention, you ora is on localhost! you can use -ora to specify ora ip"
+    time.sleep(5)
+shell_response= os.system("ping -c 1 " + arg_ora )
+
+#and then check the response...
+if shell_response == 0:
+  print arg_ora, 'is up!'
+else:
+  print arg_ora, 'is down! Please check you network config!'
+  exit(1)
+
+NODECATALOG = node_select()
+
+for NODE in NODECATALOG:
+    if rackhdapi('/api/2.0/nodes/' + NODE)['json']['name'] != "Management Server":

```

Comment:
The node name can be any arbitrary words, so it's nonsense to check the node name.